CONTRIBUTING.md

```diff
@@ -1,86 +1,187 @@
+
+If you want to work on any of these issues, just drop a message saying "I'd like to work on this", and we will assign the issue to you and update the issue's status as "claimed".
+
+Alternatively, when opening an issue, you can also click the "self service" checkbox to indicate that you'd like to work on the issue yourself, which will also make us see the issue as "claimed".
+
+## Development
+
+### Online one-click setup for contributing
+
+You can also try using the new [github.dev](https://github.dev/pact-foundation/pact-python) feature. While you are browsing any file, changing the domain name from `github.com` to `github.dev` will turn your browser into an online editor. You can start making changes and send pull requests right away.
+
+### Installation
+
+1.  Ensure you have [Python](https://www.python.org/) installed.
+
+2.  Ensure you have [Hatch](https://hatch.pypa.io/) installed. This is used to manage the development environment can be installed as follows:
+
+    ```sh
+    python -m pip install --user pipx # If you don't have pipx
+    pipx install hatch
+    ```
+
+3.  After cloning the repository, run `hatch shell` in the root of the repository. This will install all dependencies in a Python virtual environment and then ensure that the virtual environment is activated.
+
+4.  To run tests, run `hatch run test` to make sure the test suite is working. You should also make sure the example works by running `hatch run example`. For the examples, you will have to make sure that you have Docker (or a suitable alternative) installed and running.
+
+5.  If you want to run the tests against all supported Python versions, you can run `hatch run test:all`.
+
+### Code Conventions
 
-Run the tests:
+-   **Most important: Look around.** Match the style you see used in the rest of the project. This includes formatting, naming files, naming things in code, naming things in documentation, etc.
+-   "Attractive"
+-   We do have Black (a formatter) and Ruff (a syntax linter) to catch most stylistic problems. If you are working locally, they should automatically fix some issues during git commits and push.
+
+Don't worry too much about styles in general—the maintainers will help you fix them as they review your code.
+
+To help catch a lot of simple formatting or linting issues, you can run `hatch run lint` to run Black and Ruff. This process can also be automated by installing [`pre-commit`](https://pre-commit.com/) hooks:
+
+```sh
+pre-commit install
 ```
-make test
+
+## Pull Requests
+
+So you have decided to contribute code back to upstream by opening a pull request. You've invested a good chunk of time, and we appreciate it. We will do our best to work with you and get the PR looked at.
```

Comment:
they may not have invested the time yet and are just reading the docs on how to contribute.

Making a pull request can often feel daunting but it should need to be. We are appreciate anyone experimenting with the code, but we greatly appreciate upstream changes as these assist maintainers and can deliver real value to end users.

I would maybe put something in about considering if the change it useful for a subset of people, or all users, and considering should be paid to documenting a feature. If someone hasn't already, is it worth them filling out a feature request issue template (rather than having to hand roll their own desc)

Also would the feature benefit just pact-python, or the wider pact ecosystem. (and considerations for the implications)

Reply:
> they may not have invested the time yet and are just reading the docs on how to contribute.

Very true! Let me fix this up and make it clear that before a PR is made, a ticket should ideally have been created first (which then guides them through the ticket workflow and iron out some kinks).